### PR TITLE
Updates to the PyTorch explainers

### PR DESCRIPTION
**Pull request by gabrieltseng**
Two updates:

1. Previously, if the input tensor passed to the explainer has `requires_grad == True`, then the DeepExplainer throws the following RunTime error:
```
Can't call numpy() on Variable that requires grad. Use var.detach().numpy() instead.
```
This is fixed by ensuring all tensors are detached when they are passed to the explainer.

2. ~The PyTorch gradient explainer is updated to handle `return_variances`.~ This is fixed in #873

### CODE REVIEW

**Comment by slundberg** [comment]
Thanks @gabrieltseng !
